strawberry/aiohttp/views.py

```diff
@@ -0,0 +1,106 @@
+        return aiohttp.web.HTTPNotFound()
+
+    async def post(self) -> aiohttp.web.Response:
+        operation_context = await self.get_execution_context()
+        context = await self.get_context()
+        root_value = await self.get_root_value()
+
+        result = await self.schema.execute(
+            query=operation_context.query,
+            root_value=root_value,
+            variable_values=operation_context.variables,
+            context_value=context,
+            operation_name=operation_context.operation_name,
+        )
+
+        response_data = await self.process_result(result)
+        return aiohttp.web.json_response(response_data)
+
+    async def get_execution_context(self) -> ExecutionContext:
+        try:
+            data = await self.parse_body()
+        except json.JSONDecodeError:
+            raise aiohttp.web.HTTPBadRequest(
+                reason="Unable to parse request body as JSON"
+            )
+
+        try:
+            query = data["query"]
+            variables = data.get("variables")
+            operation_name = data.get("operationName")
+        except KeyError:
+            raise aiohttp.web.HTTPBadRequest(
+                reason="No GraphQL query found in the request"
+            )
+
+        return ExecutionContext(
+            query=query,
+            variables=variables,
+            operation_name=operation_name,
+        )
+
+    async def parse_body(self) -> dict:
+        if self.request.content_type.startswith("multipart/form-data"):
+            reader = await self.request.multipart()
+            operations = {}
+            files_map = {}
+            files = {}
+            async for field in reader:
+                if field.name == "operations":
+                    operations = await field.json()
+                elif field.name == "map":
+                    files_map = await field.json()
+                elif field.filename:
+                    files[field.name] = BytesIO(await field.read(decode=False))
+            return replace_placeholders_with_files(operations, files_map, files)
+        return await self.request.json()
+
+    def render_graphiql(self) -> aiohttp.web.Response:
+        with open(self.graphiql_html_file_path, "r") as f:
+            html_string = f.read()
+        html_string = html_string.replace("{{ SUBSCRIPTION_ENABLED }}", "false")
+        return aiohttp.web.Response(text=html_string, content_type="text/html")
+
+    @property
+    def graphiql_html_file_path(self) -> Path:
+        return Path(__file__).parent.parent / "static" / "graphiql.html"
+
+    @property
+    def should_render_graphiql(self) -> bool:
+        if not self.graphiql:
+            return False
+        return "text/html" in self.request.headers.get("Accept", "")
+
+    @classmethod
+    def as_view(cls, schema: BaseSchema, graphiql=True) -> Type["GraphQLView"]:
+        cls.schema = schema
+        cls.graphiql = graphiql
+        return cls
```

Comment:
This modifies _and_ returns the global object. Is that intentional?

Reply:
It somewhat is. `aiohttp.web.Application.router.add_view` expects a subclass of `aiohttp.web.View` as parameter. The `as_view` method is what I came up with to make the subclass configurable and more consistent with the other integrations.

The class based view appeared to be the easiest for the user to configure routing for. I could explore two alternatives to class based views which would not depend on my somewhat hacky configuration approche.

Reply:
Would it not be unreasonable for auser to subclass the `aiohttp` `GraphQLView` object to override those two values?

Reply:
I thought about that. Creating a subclass would be required since the abviously `schema` is not optional. Requiring the user to create a subclass in order to configure it felt incovenient to me. It's an option though.

I just realized that I could probably achive something like the following by not using a class based view and utilizing a `__call__` method:

```python
app = aiohttp.web.Application()
app.add_routes([web.route('*', '/graphql', GraphQLView(schema=schema, graphiql=False))])
```

I'll give this approche a try.

Reply:
I manage to implement the solution mentioned in the comment before. Feel free to take a look :)